### PR TITLE
#159206062  Users can search for articles

### PR DESCRIPTION
**Pull request by klevamane**
#### What does this PR do?
- Search for articles by tag, title, and author
#### Description of Task to be completed?
- write units test for the response searching for articles
- return results of searched articles if found

#### How should this be manually tested?
Have the following endpoints working

 GET /api/articles/?tag=tagname

GET /api/articles/?title=title

GET /api/articles/?author=username

 In postman navigate to `http://localhost:3000/api/users`
  
#### Any background context you want to provide?
- Pull in changes and run `npm install`
- Start server with `npm run dev`

#### What are the relevant pivotal tracker stories?
#159206049

[Delivers #159206062]

### CODE REVIEW

**Review by OKiMaureen — APPROVED** [review]

**Review by Oloyedesinmiloluwa — APPROVED** [review]

File: helpers/exports.js [hunk 1]
```diff
@@ -35,3 +39,66 @@ exports.resendVerificationEmail = async (req, res) => {
...
+    };
+  }
+  /**
+  * set the relationship to search from the right table
+  * since the attribute to be searched for is on the right table
+  * */
+  if (searchCriteria === 'author') {
+    searchParameters.where = {
+      [Op.or]:
+        [{ firstname: Sequelize.where(Sequelize.col('User.firstname'), { [Op.eq]: `${textToSearch}` }) },
```
Comment (anuonifade) [inline helpers/exports.js]: I think it will be better to use `iLike` here as well

File: helpers/exports.js [hunk 2]
```diff
@@ -35,3 +39,66 @@ exports.resendVerificationEmail = async (req, res) => {
...
+          { lastname: Sequelize.where(Sequelize.col('User.lastname'), { [Op.eq]: `${textToSearch}` }) }
```
Comment (anuonifade) [inline helpers/exports.js]: Same here

**Review by anuonifade — CHANGES_REQUESTED** [review]
Please address my comment

**Review by anuonifade — APPROVED** [review]
LGTM

**Pull request merged**